avm/res/event-grid/system-topic/main.bicep

```
@@ -85,15 +88,25 @@ var builtInRoleNames = {
 
 var formattedRoleAssignments = [
   for (roleAssignment, index) in (roleAssignments ?? []): union(roleAssignment, {
-    roleDefinitionId: builtInRoleNames[?roleAssignment.roleDefinitionIdOrName] ?? (contains(
-        roleAssignment.roleDefinitionIdOrName,
+    roleDefinitionId: builtInRoleNames[?roleAssignment.roleDefinitionId] ?? (contains(
+        roleAssignment.roleDefinitionId,
         '/providers/Microsoft.Authorization/roleDefinitions/'
       )
-      ? roleAssignment.roleDefinitionIdOrName
-      : subscriptionResourceId('Microsoft.Authorization/roleDefinitions', roleAssignment.roleDefinitionIdOrName))
+      ? roleAssignment.roleDefinitionId
+      : subscriptionResourceId('Microsoft.Authorization/roleDefinitions', roleAssignment.roleDefinitionId))
```

```suggestion
      ? roleAssignment.roleDefinitionIdOrName
      : subscriptionResourceId('Microsoft.Authorization/roleDefinitions', roleAssignment.roleDefinitionIdOrName))
```

The default role assignment implementation must remain the same